### ISSUE TITLE
instal.packages('anything') just hangs forever

### ISSUE THREAD

**Issue by statquant** [Enchufa2/bspm#19]
Hello, I gave a try to bspm without success.
Here is what I did

I install bspm as root
```
cd /tmp
wget https://cloud.r-project.org/src/contrib/bspm_0.3.5.tar.gz
sudo R CMD INSTALL bspm_0.3.5.tar.gz
```
=> that worked fine

Then in R
```
R --vanilla
library(bspm)
bspm::enable()
install.packages('here')
```
=> there is hangs forever
```

> sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Fedora 32 (Workstation Edition)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblas-r0.3.9.so

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
 [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
 [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] bspm_0.3.5

loaded via a namespace (and not attached):
[1] compiler_3.6.3
```

**Comment by Enchufa2** [comment]
This is SELinux blocking your manually-installed package. If you disable SELinux, it will work. But I don't recommend it. I'll package `bspm` for Fedora so that it can be installed from the system repos and SELinux won't protest then.

Anyway, my recommendation is to use the [iucar/cran](https://copr.fedorainfracloud.org/coprs/iucar/cran/) Copr repo. The `CoprManager` package, included there, is just a "rebranding" of `bspm`, tailored specifically to manage that repo.

**Comment by statquant** [comment]
Ok, thanks, unless I misunderstand something my setup is vanilla so it might be worth addin a line in the REAME.md.
I just tested iucar/cran and that's truely amazing, thank you.
I have a question on install.packages, I assume that because dnf is called under the hood there is no way the instaled package in installed in the USER_LIBS path, but I just wanted to check. If it could be done that would be a huge plus as it could also be used in an "enterprise" setup where you do not have access to root.

**Comment by Enchufa2** [comment]
> Ok, thanks, unless I misunderstand something my setup is vanilla so it might be worth addin a line in the REAME.md.

Sure, I'll add a warning about SELinux...

> I just tested iucar/cran and that's truely amazing, thank you.

... and about this. With iucar/cran and CoprManager, you have access to almost the whole CRAN, so IMO there's no point in installing `bspm` from CRAN on Fedora.

> I have a question on install.packages, I assume that because dnf is called under the hood there is no way the instaled package in installed in the USER_LIBS path, but I just wanted to check.

No, there isn't. 

> If it could be done that would be a huge plus as it could also be used in an "enterprise" setup where you do not have access to root.

Not sure if I follow. With `bspm`/`CoprManager`, you are actually installing packages in the system without root access. That's the purpose of the package.

**Comment by statquant** [comment]
Hello, with regards to root I mean that for me the packages installed with CoprManager are installed in /usr/local/lib/R/library and are owned by root. I do not think that would be something I'd be able to do at work.

**Comment by Enchufa2** [comment]
Yes, and another advantage of this is that a system with e.g. 15 users don't have 15 installations of probably outdated versions of hundreds of packages. I think this is an advantage for sysadmins: if they activate the Copr repo and enable CoprManager, all users will be installing (without root privileges) and sharing packages under /usr/local/lib/R/library, and the sysadmin can ensure that they are up-to-date.

**Comment by statquant** [comment]
True but at the same time I think most companies will have parallel install
for different version of R. I do not think you want people sharing
libraries ie. some users updating packages that other would like to keep as
is.
Best setup (in my view) is multiple installs with large system libraries
(per version) shared by multiple users and local libraries for personal
use. But using bspm in this setup would require the possibility of
specifying some kind of —prefix for installation.

**Comment by Enchufa2** [comment]
Not really: `bspm` does not decide the installation directory; R does. So it is already compatible with the setup you are describing.

**Comment by statquant** [comment]
Already compatible with this setup if not used by non admin users right (as
bspm will not install in USER_LIBS) ? Else I am not following :\

**Comment by Enchufa2** [comment]
I am saying is that, if you have multiple R installations from multiple system repos, each repo with its own packages and stuff, then bspm can pull them out in their different installation paths.

What you cannot expect, of course, is that _from a single system repository_, anyone would be able to pull out those binary packages to different instalation directories **and** magically convert them to different versions of R. But that's not bspm's limitation, that's how system repositories work.

**Comment by statquant** [comment]
Ok sure that’s well understood.
Regards

**Comment by Enchufa2** [comment]
(BTW I'm keeping this open for me to remember to add something to the README. Thanks for the discussion!)
